addons/mail/static/src/components/message/message.xml

```diff
@@ -76,9 +94,12 @@
                                     Anonymous
                                 </div>
                             </t>
-                            <div class="o_Message_date o_Message_headerDate" t-att-class="{ 'o-message-selected': isSelected }" t-att-title="datetime">
-                                - <t t-esc="message.dateFromNow"/>
-                            </div>
+                            <t t-if="!message.hasSendError">
```

Comment:
Wondering: shouldn't pending/error messages not have a date, and condition being on message having actually a date?